bundle/config/mutator/resolve_resource_references.go

```diff
@@ -0,0 +1,61 @@
+package mutator
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/resolvers"
+	"github.com/databricks/cli/libs/log"
+)
+
+var separator string = ":"
+
+type resolveResourceReferences struct {
+	resolvers map[string](resolvers.ResolverFunc)
+}
+
+func ResolveResourceReferences() bundle.Mutator {
+	return &resolveResourceReferences{
+		resolvers: resolvers.Resolvers(),
+	}
+}
+
+func (m *resolveResourceReferences) Apply(ctx context.Context, b *bundle.Bundle) error {
+	for k, v := range b.Config.Variables {
+		if v.Lookup == "" {
+			continue
+		}
+
+		if v.HasValue() {
+			log.Debugf(ctx, "Ignoring '%s' lookup for the variable '%s' because the value is set", v.Lookup, k)
+			continue
+		}
+
+		lookup := v.Lookup
+		parts := strings.Split(lookup, separator)
```

Comment:
Can use `strings.Cut` to force 2 output chunks (also allows use of `:` in the name).